src/app/UploadPilets.tsx

```diff
@@ -1,23 +1,31 @@
 import { FC, ChangeEvent, useState } from 'react';
-import { CustomInput, InputGroup, InputGroupAddon, Button } from 'reactstrap';
+import { CustomInput, InputGroup, InputGroupAddon, Button, Progress, UncontrolledAlert } from 'reactstrap';
 import { jsx } from '@emotion/core';
 import { injectPiletsFromUrl } from './utils';
 
-export interface UploadPiletsProps {}
+export interface UploadPiletsProps { }
 
 export const UploadPilets: FC<UploadPiletsProps> = () => {
   const [file, setFile] = useState({
     value: undefined,
     key: 0,
   });
 
+  const [progress, setProgress] = useState(0);
+  const [failedUpload, setFailedUpload] = useState(false);
```

Comment:
What's the difference between `failedUpload` and `err`? Can't we use just one?

Also you use 4 states where a single one would be sufficient (they are all linked).

Reply:
failedUpload it is only a boolean item to keep tracking if the Loading Bar is opened or closes.

err => is a text of error that we display if the upload was not successful.

Reply:
Well, then the name `failedUpload` is not really well taken, right?